src/commands/aksCompareCluster/aksCompareCluster.ts

```diff
@@ -0,0 +1,123 @@
+import { QuickPickItem } from 'vscode';
+import { createQuickPickStep, runMultiStepInput } from '../utils/multiStepHelper';
+import { IActionContext } from "@microsoft/vscode-azext-utils";
+import { getAksClusterSubscriptionItem, getContainerClient, getResourceManagementClient, } from '../utils/clusters';
+import { failed } from '../utils/errorable';
+import * as vscode from 'vscode';
+import * as k8s from 'vscode-kubernetes-tools-api';
+import SubscriptionTreeItem from '../../tree/subscriptionTreeItem';
+import { getExtension } from '../utils/host';
+import { Dictionary } from '../utils/dictionary';
+import * as tmpfile from "../utils/tempfile";
+import { longRunning } from "../utils/host";
+
+interface State {
+    clusterGroupCompareFrom: ClusterNameItem;
+    clusterGroupCompareWith: ClusterNameItem;
+    clustername: string;
+    subid: string | undefined;
+}
+
+interface ClusterNameItem extends QuickPickItem {
+    name: string;
+}
+
+/**
+ * A multi-step input using window.createQuickPick() and window.createInputBox().
+ * 
+ * This first part uses the helper class `MultiStepInput` that wraps the API for the multi-step case.
+ */
+export default async function aksCompareCluster(
+    _context: IActionContext,
+    target: unknown
+): Promise<void> {
+    const cloudExplorer = await k8s.extension.cloudExplorer.v1;
+
+    const cluster = getAksClusterSubscriptionItem(target, cloudExplorer);
+    if (failed(cluster)) {
+        vscode.window.showErrorMessage(cluster.error);
+        return;
+    }
+    const subscriptionTreeItem = <SubscriptionTreeItem>cluster.result;
+
+    const extension = getExtension();
+    if (failed(extension)) {
+        vscode.window.showErrorMessage(extension.error);
+        return;
+    }
+    const containerServiceClient = getContainerClient(subscriptionTreeItem);
+    const clusterList: string[] = [];
+    const clusterGroupList: Dictionary<string> = {};
+
+    await longRunning(`Getting AKS Cluster list for ${subscriptionTreeItem.name}`, async () => {
+        const iterator = containerServiceClient.managedClusters.list();
+        for await (const cluster of iterator.byPage()) {
+            clusterList.push(...cluster.map(c => c.name ?? ''));
```

Comment:
I don't think I actually knew about the `??` operator!

There is a slight inconsistency here, though: on line 55 you handle the `name` property being null or undefined, but on line 59 you assert it's non-null with the `!` operator.

Minor thing: I prefer to use `Array.forEach` instead of `map` for operations that perform mutations. It doesn't really matter, but I think `map` is more typically used for immutable operations that return a value.

Oh, and I prefer the name `clusterResourceDictionary` that you used below.

We could do something like:
```typescript
const clusterResourceDictionary: Dictionary<string> = {};
for await (const clusters of iterator.byPage()) {
    const validClusters = clusters.filter(c => c.id && c.name);
    clusterList.push(...validClusters.map(c => c.name!);
    validClusters.forEach(c => {
        clusterResourceDictionary[c.name!] = c.id!;
    });
}
```